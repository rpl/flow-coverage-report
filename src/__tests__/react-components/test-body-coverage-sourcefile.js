'use babel';

import React from 'react';
import renderer from 'react-test-renderer';

import {FLOW_COVERAGE_SUMMARY_DATA} from '../fixtures';

import {BASE_DIR} from './common';

const REACT_COMPONENT = `${BASE_DIR}/body-coverage-sourcefile`;

test('<HTMLReportBodySourceFile />', () => {
  const HTMLReportBodySourceFile = require(REACT_COMPONENT).default;
  const fileName = 'src/a.js';
  const props = {
    coverageSummaryData: FLOW_COVERAGE_SUMMARY_DATA,
    coverageData: FLOW_COVERAGE_SUMMARY_DATA.files[fileName],
    summaryRelLink: '../index.html',
    fileName
  };

  const tree = renderer.create(<HTMLReportBodySourceFile {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test.todo('<HTMLReportBodySourceFile /> with missing props');
