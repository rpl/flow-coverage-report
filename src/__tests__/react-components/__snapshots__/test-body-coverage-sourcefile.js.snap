// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HTMLReportBodySourceFile /> 1`] = `
<body>
  <div
    className="ui grid container"
  >
    <div
      className="row"
    >
      <div
        className="twelve wide column"
      >
        <h2
          className="ui header"
        >
          <a
            href="../index.html"
            id="link-to-summary"
          >
            Flow Coverage Report
          </a>
        </h2>
      </div>
    </div>
    <div
      className="row"
    >
      <table
        className="ui small celled unstackable table"
      >
        <thead>
          <tr>
            <th
              data-sort-default={true}
            >
              Filename
            </th>
            <th>
              Annotation
            </th>
            <th>
              Percent
            </th>
            <th>
              Total
            </th>
            <th>
              Covered
            </th>
            <th>
              Uncovered
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="positive"
          >
            <td
              className=""
            >
              src/a.js
            </td>
            <td>
               @
              flow
               
            </td>
            <td
              className=""
            >
              <span>
                50
                 %
              </span>
            </td>
            <td>
               
              2
               
            </td>
            <td>
               
              1
               
            </td>
            <td>
               
              1
               
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="row ui one column centered grid"
    >
      <div
        className="column"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="ui form"
          >
            <div
              className="fields"
            >
              <div
                className="sixteen wide inline field"
              >
                <select
                  className="ui search dropdown uncovered-locations"
                >
                  <option
                    value=""
                  >
                    Uncovered Locations
                  </option>
                  <option
                    value={1}
                  >
                    Start: 1,1 - End: 2,2
                  </option>
                </select>
              </div>
              <div
                className="four wide inline field"
              >
                <select
                  className="ui search dropdown syntax-highlighting"
                >
                  <option
                    value="es"
                  >
                    ES6/ES7
                  </option>
                  <option
                    value="js"
                  >
                    JavaScript
                  </option>
                  <option
                    value="no"
                  >
                    None
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <textarea
          id="file-content"
          readOnly={true}
          value="undefined"
        />
      </div>
    </div>
    <div
      className="row centered"
    >
      <footer>
        Flow Coverage Report generated by
        <a
          href="https://flowtype.org"
        >
           flow 
        </a>
        and
        <a
          href="https://github.com/rpl/flow-coverage-report"
        >
           flow-coverage-report 
        </a>
        at 
      </footer>
    </div>
  </div>
  <pre
    id="file-coverage-data"
    style={
      Object {
        "display": "none",
      }
    }
  >
    {"percent":50,"annotation":"flow","isFlow":true,"expressions":{"covered_count":1,"uncovered_count":1,"uncovered_locs":[{"start":{"line":1,"column":1,"offset":10},"end":{"line":2,"column":2,"offset":30}}]}}
  </pre>
</body>
`;
